### ISSUE TITLE
SourceKit Language Server server crashes

### ISSUE THREAD

**Issue by LazyOtto** [apple/sourcekit-lsp#635]
Repeatable crash when used with VSCode on Linux Mint.
Installed with: swift-5.7-RELEASE-ubuntu22.04.tar.gz
Code to cause it to crash - second line below:
```
var someInts: [Int] = []
someInts.append()
// someInts.append(contentsOf: [5, 7, 9])
print("An array of \(someInts)) with count of: \(someInts.count)")
```
Stack trace of thread 29377:
#0  0x0000555b412ae48e n/a (/home/user/Applications/swift-5.7-RELEASE-ubuntu22.04/usr/bin/sourcekit-lsp + 0x9e648e)

```
Core was generated by `sourcekit-lsp'.
Program terminated with signal SIGILL, Illegal instruction.
#0  Swift runtime failure: precondition failure () at /home/build-user/sourcekit-lsp/Sources/SourceKitLSP/Swift/Diagnostic.swift:83
83	/home/build-user/sourcekit-lsp/Sources/SourceKitLSP/Swift/Diagnostic.swift: No such file or directory.
[Current thread is 1 (Thread 0x7f2b92cf7640 (LWP 29377))]
```

**Comment by LazyOtto** [comment]
I replaced the precondition check in "fixitTitle" as below. Have not run any suite of tests on the patch other than the case which I stumbled upon above.
```
  public static func fixitTitle(replace oldText: String, with newText: String) -> String {
    switch (oldText.isEmpty, newText.isEmpty) {
    case (false, false):
      return "Replace '\(oldText)' with '\(newText)'"
    case (false, true):
      return "Remove '\(oldText)'"
    case (true, false):
      return "Insert '\(newText)'"
    case (true, true):
      // preconditionFailure("FixIt makes no changes")
      log("FixIt makes no changes", level: .warning)
      return ""
    }
  }
```

**Comment by LazyOtto** [comment]
Tests supplied with the package pass:
```
Test Suite 'All tests' passed at 2022-09-17 10:22:47.836
Executed 249 tests, with 4 tests skipped and 0 failures (0 unexpected) in 32.119 (32.119) seconds
```

**Comment by ahoppen** [comment]
rdar://100158700

**Comment by ahoppen** [comment]
I suspect you are using an editor that has snippets disabled, in which case this issue should be fixed by https://github.com/apple/sourcekit-lsp/pull/602.

To verify that this is indeed the case could you:
- Check if this issue still happens with a recent main development snapshot.
- Check if snippets are enabled in your VSCode
  - In VSCode open the Output windows (Command Palette -> Output: Focus on Output View) -> select SourceKit Language Server and paste the `initialize` request here (should be the first 500-ish lines). I suspect that it has `snippetSupport` set to `false`.

**Comment by LazyOtto** [comment]
Tests supplied with the package pass:
```
Test Suite 'All tests' passed at 2022-09-17 10:22:47.836
Executed 249 tests, with 4 tests skipped and 0 failures (0 unexpected) in 32.119 (32.119) seconds
```

**Comment by LazyOtto** [comment]
I will try the suggested test later.

More information for now, editor/VSCode version in use is:
```
Version: 1.71.2
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:12:14.256Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Linux x64 5.15.0-47-generic
Sandboxed: No
```

Searching on "snippets" in Settings shows seven relevant hits. (Four for "Text Editor" and three for "Emmet".)

All are enabled *except* "Extensions:Emmet Show Suggestions as Snippets".

**Comment by adam-fowler** [comment]
Hi guys, 
I can replicate this issue, and I have snippetSupport set to true. Also I have reduce the amount of text you need to get the crash to 
```swift
var a: [Int] = []
a.append()
```
This does appear to be fixed on main. If we could bring what fixed it across to the 5.7 branch that would be great. The above code seems pretty common place.

**Comment by ahoppen** [comment]
Ah, I see what’s happening now. We disable snippet support when translating the insertion response for code completion (because LSP doesn’t support snippets for code completion). I opened a PR to get this fixed in 5.7 but can’t make any promises when this will end up in a toolchain release. https://github.com/apple/sourcekit-lsp/pull/646
